src/targets/gpu/device/include/migraphx/gpu/device/float_equal.hpp

```diff
@@ -39,7 +39,11 @@ using common_type = typename std::common_type<Ts...>::type;
 template <class T, MIGRAPHX_REQUIRES(is_floating_point<T>{})>
 __device__ bool float_equal_device(T x, T y)
 {
-    return std::isfinite(x) and std::isfinite(y) and
+    // The standard library from MSVC does not implement std::isfinite() for integer
+    // types - no additional overloads are provided. According to the documentation,
+    // integer types should be treaded as doubles.
+    // Refer to https://en.cppreference.com/w/cpp/numeric/math/isfinite for more information.
+    return std::isfinite(static_cast<double>(x)) and std::isfinite(static_cast<double>(y)) and
```

Comment:
type `T` is reuqired to be floating point for this function to be called.

Reply:
Yes, I didn't see type T. I'll revert these changes.